web/src/ClearLogs.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react"
+import styled from "styled-components"
+import { Alert } from "./alerts"
+import { FilterSource } from "./logfilters"
+import { isBuildSpanId, isTiltfileSpanId } from "./logs"
+import { useLogStore } from "./LogStore"
+import { Color, FontSize } from "./style-helpers"
+
+const ClearLogsButton = styled.a`
+  margin-left: auto;
+  cursor: pointer;
+  font-size: ${FontSize.small};
+
+  &:hover {
+    color: ${Color.blue};
+  }
+`
+
+export interface ClearLogsProps {
+  resource?: Proto.webviewResource
+  alerts?: Alert[]
+}
+
+const ClearLogs: React.FC<ClearLogsProps> = ({ resource, alerts }) => {
+  const logStore = useLogStore()
+  const label = resource?.name ? "Clear Logs" : "Clear All Logs"
+
+  const clearLogs = () => {
+    let spans: { [key: string]: Proto.webviewLogSpan }
+    let relevantAlerts: Alert[]
+    if (resource) {
+      const manifestName = resource.name ?? ""
+      spans = logStore.spansForManifest(manifestName)
+      relevantAlerts = (alerts || []).filter(
+        (alert) => alert.resourceName == manifestName
+      )
+    } else {
+      spans = logStore.allSpans()
+      relevantAlerts = alerts || []
+    }
+
+    // don't delete the latest span that corresponds to an alert
```

Comment:
you've introduced a new concept here of a "latest" span. I don't think this concept makes sense. 

You adapted it from ordered build spans, which was already sus. But at least ordered build spans corresponds to something real in the underlying engine - a manifest can only have one build at a time. But I don't think this is true for arbitrary spans for things like pod logs and events

might make sense to take a step back and think about what we're trying to do and the right way to model it?

Reply:
the special-casing of tiltfile span ids is a pretty big red flag to me that we may need to talk more about the model of what you're trying to do here

but i'm not 100% clear on why we need to keep the last alert at all?

Reply:
The intent here was to avoid getting into a state where the alerts bar + log filters are showing errors/warnings but there's no longer any content.

Example:
<img width="1005" alt="Screen Shot 2021-02-08 at 1 14 50 PM" src="https://user-images.githubusercontent.com/841263/107263693-0fadc700-6a10-11eb-81ba-8d423bab3eeb.png">

(That's after hitting "Clear Logs" in the current state of this branch.)

The runtime case is definitely weirder given that it only applies to K8s and as shown here, it's unlikely the log output is useful to diagnose the problem anyway.

Arguably, this is a pre-existing issue as alerts & log output are decoupled as-is -- alerts actually have a message but afaict it's not shown anywhere, so it's already possible to have the log filter suggesting there are errors/warnings but none actually exist in the log output.

Reply:
I'll simplify to removing _all_ spans for manifest rather than trying to keep around the related error context. I do think the way alerts are presented / interact with log filters is fairly unintuitive when using the UI and clearing logs can/might make that slightly worse, but at least it's an intentional action so hopefully not hugely surprising

Reply:
ah, ok. so if I understand this correctly - you're trying to going through the alerts, and trying to preserve the logs that keep those alerts on screen?

what if we did the opposite? go through the logs that we're clearing, and add a flag to indicate "ignore alerts in span X before time Y"?

Reply:
Okay, take a look at latest changes. Since the alert logic already has span info, it was easier to filter out alerts that are for a span that doesn't exist in the log store rather than doing additional book keeping by timestamp.